### ISSUE TITLE
Ensure the migration from master to version 2 will work

### ISSUE THREAD

**Issue by mconf-daileon** [mconf/mconf-web#507]
---

Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Redmine Issue: 987, http://dev.mconf.org/redmine/issues/987

Original Assignee: Leonardo Daronco

---

Some tables and columns were added both on @master@ and on @branch-v2@. We have to check everything to make sure the database will be consistent and the migrations will work when migrating from @master@ to @branch-v2@.
This should be tested also using real databases from version 1.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2013-10-02T01:26:23Z

---

The migrations were adjusted and we tested the migration from the current @production@ branch (2859eb925dd00f3198538d17a9957c7a4481cedc) to the current @branch-v2@ (49389712ddd7150842e0f4879836967aced83a33) and everything is working. Tested migrating the database of mconf.org as of 01/Oct/2013.

Our internal capistrano scripts were updated, but "Mconf-Web's public caphub":https://github.com/mconf/mconf-web-caphub still not up-to-date.

We still have to test:

* Install a new application from scratch on the new version using "the new manual":https://github.com/mconf/mconf-web/wiki/Mconf-Web-Deployment-Manual-v2 (still in construction)
* Install a new application in the current version (0.8) following "the current manual":https://github.com/mconf/mconf-web/wiki/Mconf-Web-Deployment-Manual (without capistrano) and then migrate it to the new version

Other things to be considered:

* Recent activity: The way recent activities are found and shown changed, and now, after migrating, the recent activity for all users will be "lost". We have to consider if we want to migrate them somehow or if the way it is now is ok.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2014-01-21T15:28:45Z

---

A big chunk of the work was already done. The databases of a few production servers were used to test the migration and it went ok.
Leaving it open because if has to be reviewed again once the new version is done, since more things will change until then.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2014-05-15T15:06:26Z

---

I've seen several errors when running the migration "BigbluebuttonRailsTo140C". Most of them are one or more of the following:

* "Name is too long (maximum is 150 characters)"
* "Name has already been taken"
* "String ID has already been taken"
* "Attendee Password is too long (maximum is 16 characters)"

We have to solve them otherwise the rooms will be without an associated @room_options@ and it can break things.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2014-08-04T20:20:03Z

---

The application is at a point where the database is mostly as it will be when v2 is released. The changes from now on should be small.


There are several possible inconsistencies in the database that will appear after the migration, not only the ones in "BigbluebuttonRailsTo140C". I'm currently cleaning up the migrations to make sure they will run properly, even if a few things are left inconsistent. The migrations should deal only with their target data, and not try to solve all possible inconsistencies in the models.

The inconsistencies should be solved in some other way. There is a gem called "active_sanity" that checks for inconsistencies in the db and prints them. We could use it to code a separate script that deals with these inconsistencies, trying to solve as many as possible automatically. For the ones that can't be solved automatically we should at least show them to the user and point a possible solution.
This should then be offered as an additional step in the migration to v2, to be executed after the database migration.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-04-07T17:39:41Z

---

Most of the hard work was already finished and tested. The migration from 0.8 to the current state of 2.0 should be working as expected.
A wiki page was created with all the steps necessary to migrate: https://github.com/mconf/mconf-web/wiki/Migration-v2

From now on we just have to make sure the migration from the current state of 2.0 to future states until 2.0 is released is ok.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-04-13T14:02:56Z

---

When tested again in a migration from 0.8.1 to master@3e0080e7f07, two migrations failed.
One was already fixed (see https://github.com/mconf/mconf-web/commit/a10ef9b4c7d38) and the other was on @MigrateEventsToMwebEvents@, caused because the table @participant_confirmations@ didn't exist.

This ticket can only be closed when 2.0 is out, otherwise any change in the application might break previous migrations. So, when 2.0 is out and the migration is tested, this can be closed.

**Comment by mconf-daileon** [comment]
---

Original Redmine Comment
Author Name: **Leonardo Daronco** (Leonardo Daronco)
Original Date: 2015-10-28T15:12:20Z

---

Tested again with a database from 0.8 and everything went fine. Just tested to make sure migrations were not using classes not available anymore, the actual job the migrations do were already tested a lot in the past.
